qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py

```diff
@@ -465,14 +465,95 @@ def __init__(self, name: str, instrument: B1517A, **kwargs: Any):
         self.source_voltage = _FMTResponse(None, None, None, None)
         self._fudge: float = 1.5
 
-    def get_raw(self) -> Tuple[Tuple[float, ...], Tuple[float, ...]]:
+    def set_names_labels_and_units(self,
+                                   names: Optional[Sequence[str]] = None,
+                                   labels: Optional[Sequence[str]] = None,
+                                   units: Optional[Sequence[str]] = None
+                                   ) -> None:
         measurement_mode = self.instrument.get_measurement_mode()
-        if len(measurement_mode['channels']) != 2:
-            raise ValueError('Two measurement channels are needed, one for '
-                             'gate current and other for source drain '
-                             'current.')
+        channels = measurement_mode['channels']
 
-        smu = self.instrument.by_channel[measurement_mode['channels'][0]]
+        if names is None:
+            names = [f"param{n+1}" for n in range(len(channels))]
+            if labels is None:
+                labels = [f"Param{n + 1} Current" for n in range(len(channels))]
+
+        if labels is None:
+            labels = tuple(names)
+
+        if units is None:
+            units = ['A'] * len(names)
+
+        if len(labels) != len(names) or len(units) != len(names):
+            raise ValueError(
+                f"If provided, the number of names, labels, and units must be "
+                f"the same, instead got {len(names)} names, {len(labels)} "
+                f"labels, {len(units)} units."
+            )
+
+        if len(names) != len(channels):
+            raise ValueError(
+                f"The number of names ({len(names)}) does not match the number "
+                f"of channels expected for the IV sweep measurement, "
+                f"which is {len(channels)}. Please, when providing names, "
+                f"provide them for every channel."
+            )
+
+        self.names = tuple(names)
+        self.labels = tuple(labels)
+        self.units = tuple(units)
+
+        for n in range(len(channels)):
+            setattr(self, f"param{n+1}", _FMTResponse(None, None, None, None))
+
+        self.shapes = ((1,),) * len(self.names)
+
+        self.set_setpoint_name_label_and_unit()
+
+    def set_setpoint_name_label_and_unit(
+            self,
+            name: Optional[str] = None,
+            label: Optional[str] = None,
+            unit: Optional[str] = None
+    ) -> None:
+        # number of measured parameters of this MultiParameter
+        n_names = len(self.names)
+
+        name = name if name is not None else "Voltage"
```

Comment:
Similarly here if the setpoint specifics aren't specified then the default mode is for souring voltage right? That seems fine to me but maybe add a quick note about  that in the doc string.

Reply:
yes, that's actually the only way this IV sweep works, it only sweeps voltage. it's left specifyable in case someone wants to be more specific for some unknown reason :)

> add a quick note about that in the doc string

will definitely do!